addons/point_of_sale/static/src/app/generic_components/input/input.js

```diff
@@ -8,21 +8,20 @@ import { debounce } from "@web/core/utils/timing";
     This component is meant to provide a "batteries included" api for working
     with inputs. It is well suited to work as a search bar or as a monetary input.
     Optional props allow handling debouncing, toggling between mobile and desktop views,
-    autofocus, validation, and more. 
+    autofocus, validation, and more.
     The only required prop is tModel, which is an array that represents the
     state variable that this component should read and write to. The first
-    element of the array is the object returned by `useState` and the rest of 
-    the elements are strings representing the keys to access the value.
-    (they can also be numbers in the case of arrays).
+    element of the array is the object returned by `useState` and second is
+    either a `string` or an `int` representing the key to access the value.
     ex: normally you would write
     ```<input t-model="state.userValues[nThValue].searchString"/>```
     with this component you can write
-    ```<Input tModel="[state, "userValues", nThValue, "searchString"]"/>```
+    ```<Input tModel="[state.userValues[nThValue], "searchString"]"/>```
 */
 export class Input extends Component {
     static template = "point_of_sale.input";
     static props = {
-        tModel: { type: Array },
+        tModel: { type: Array, element: [Object, String, Number] },
```

Comment:
remove element. Only put until you know how to make a validator for "[Object, String | Number]"